src/charm_state.py

```diff
@@ -862,16 +859,20 @@ def aproxy_address(self) -> Optional[str]:
             assert (
                 proxy_address is not None and proxy_address.host is not None
             )  # nosec for [B101:assert_used]
-            aproxy_address = (
-                proxy_address.host
-                if not proxy_address.port
-                else f"{proxy_address.host}:{proxy_address.port}"
-            )
+            if "http" in proxy_address.host:
+                aproxy_address = proxy_address.host[7:]
+            elif "https" in proxy_address.host:
+                aproxy_address = proxy_address.host[8:]
+            else:
+                aproxy_address = proxy_address.host
+
+            if proxy_address.port:
+                aproxy_address = f"{proxy_address.host}:{proxy_address.port}"
         else:
             aproxy_address = None
         return aproxy_address
 
-    @validator("use_aproxy")
+    @field_validator("use_aproxy")  # type: ignore
```

Comment:
What is the type: ignore here please?

Reply:
It is because mypy complains as:
```
src/charm_state.py:875: error: Value of type variable "_V2BeforeAfterOrPlainValidatorType" of function cannot be "Callable[[type[ProxyConfig], bool, dict[Any, Any]], bool]"  [type-var]
```

I should put it as 
```python
    @field_validator("use_aproxy")  # type: ignore[type-var]
```

right?

Reply:
Your issue here is that `values` is not a `dict`, it is a `ValidationInfo`. Then, you can remove this exception and the next one in the function.

Reply:
I just tried that and can confirm we don't need the ignore comments when `ValidationInfo` type is used for `values`. Thanks @javierdelapuente !